dapps/satoshi-multiplayer/contract/sources/tournament.move

```diff
@@ -290,16 +382,21 @@ module contract::tournament {
         assert!(!option::is_none(&match.hash_0) && !option::is_none(&match.hash_1), EHashNotFound);
 
         let secret_hash = sha3_256(secret);
-        
-        // Figure who player is _0 or _1 and get hash value from their match.
-        let hash_value = if (tx_context::sender(ctx) == match.player_0) {
-            option::borrow(&match.hash_0)
+
+        // Figure out whether player 0 or player 1 is revealing and add their secret to game.
+        if (tx_context::sender(ctx) == match.player_0){
+            let hash_value = option::borrow(&match.hash_0);
+
+            // Make sure player's hash is the hash of their secret.
+            assert!(secret_hash == digest::sha3_256_digest_to_bytes(hash_value), ENotCorrectSecret);
+            option::fill(&mut match.secret_0, secret_hash);
         } else {
-            option::borrow(&match.hash_1)
-        };
+            let hash_value = option::borrow(&match.hash_1);
 
-        // Make sure player's hash is the hash of their secret.
-        assert!(secret_hash == digest::sha3_256_digest_to_bytes(hash_value), ENotCorrectSecret);
+            // Make sure player's hash is the hash of their secret.
+            assert!(secret_hash == digest::sha3_256_digest_to_bytes(hash_value), ENotCorrectSecret);
+            option::fill(&mut match.secret_1, secret_hash);
+        }
```

Comment:
You can assign the whole conditional block to `let hash_value` so that we don't have duplicate assertions.

Reply:
In such case we need another condition to find again the sender to be able to `option::fill` the correct `match_secret`, or at least I couldn't think any other way of doing it.